service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java

```diff
@@ -130,9 +131,10 @@ public StepResult doStep(FlightContext flightContext)
               zone,
               resource.getInstanceId());
           return StepResult.getStepResultSuccess();
-        } else if (HttpStatus.BAD_REQUEST.value() == e.getStatusCode()) {
-          // Don't retry bad requests, which won't change. Instead fail faster.
-          return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, e);
+        } else if (HttpStatus.BAD_REQUEST.value() == e.getStatusCode()
```

Comment:
minor: consider a util function (may be under GcpResourceValidationUtils or GcpResourceConstants) to parse / handle GoogleJsonResponseException

Reply:
+1

Reply:
Done